Std/Data/Rat/Basic.lean

```diff
@@ -101,8 +101,9 @@ def divInt : Int → Int → Rat
 
 @[inherit_doc] scoped infixl:70 " /. " => Rat.divInt
 
-/-- Implements "scientific notation" `123.4e-5` for rational numbers. -/
-protected def ofScientific (m : Nat) (s : Bool) (e : Nat) : Rat :=
+/-- Implements "scientific notation" `123.4e-5` for rational numbers. (This definition is
+`@[irreducible]` because you don't want to unfold it over e.g. `RatCast.ratCast`.) -/
```

Comment:
```suggestion
/-- Implements "scientific notation" `123.4e-5` for rational numbers. (This definition is
`@[irreducible]` because you don't want to unfold it. Use `Rat.ofScientific_def` instead.) -/
```
Wording copied from `Rat.mul`. (Yes, you should also add `ofScientific_def` in `Data.Rat.Lemmas`)

Reply:
(Got it, for some reason was under the impression the needed lemmas were autogenerated.) 

Does this mean we want something like `Rat.ofScientific_true_def` and `Rat.ofScientific_false_def`, or are we ok with that `if` being in the type?

Reply:
Actually the rhs of the equality even for the true version looks odd:
```
normalize m (10 ^ e) <| Nat.ne_of_gt <| Nat.pos_pow_of_pos _ (by decide)
```
...is there a better lemma we could state here?

Reply:
I pushed one possible solution, patterned off of `sub_def`, `sub_def'`; if it's not right we'll change it. :) (Also not sure if it should be `ofScientific_of_true_def` or not, or if brevity is better.)

Reply:
you should use `mkRat` instead of `normalize` in both cases. I agree that it would also be better to have separate true and false cases, and it is optional whether you want to add `ofScientific_def` itself.

Reply:
`ofScientific_true_def` and `ofScientific_false_def` seem fine to me as names.

Reply:
👍 Ok, I've removed the `normalize` version of `ofScientific_true_def` and just left the `mkRat` version. (Note that `ofScientific_false_def` just has a `Nat` on the rhs.) Also added `ofScientific_def` for completeness' sake.